### DISCUSSION TITLE
Chain-of-promises docs page confusion

### DISCUSSION THREAD

**Discussion by wigsaparelli** [agershun/alasql#1895] [General]
[This](https://github.com/AlaSQL/alasql/wiki/promise#chain-of-promises) section of the docs states at the top of the section:

**If you put more than one SQL commands in the same string they will (until its fixed) run sync within in the async call to alasql - so a command like "INSERT ...; SELECT * ..." might not give the expected result as the select might run before the insert is done.**

Reading this many times and looking at the code I don't think it's right.  Yes a string containing multiple SQL commands will run sync within the async call but those sync executed commands will run in order.  If indeed they are run sync (I believe they are looking at the code) then a comand like "INSERT ...; SELECT * ..." would always run the INSERT before the SELECT.

Am I missing something?  It also states "(until its fixed)" but I don't think there is anything to fix here, it's as expected or perhaps something has been "fixed" and the docs not updated...

While it's being reviewed I spotted a few typos within the docs paragraph in question:
- "more than one SQL commands" should be "more than one SQL command"
- "until its fixed" should be "until it's fixed"
- "run sync within in the async call" should be "run sync within the async call"

Thank you

**Comment by mathiasrw** [comment]
Good input! I corrected things and made the problem a bit more clear. Please see if it makes more sense now.

**Reply by wigsaparelli** [reply]
Thanks although I'm still unclear how "the second command might start before the first command is done" when the commands are "run **sync** within the async call"

**Reply by mathiasrw** [reply]
For 

<img width="471" alt="image" src="https://github.com/AlaSQL/alasql/assets/1063454/5ea88d63-9918-4ddf-ae21-9d74d2f77602">

AlaSQL needs to be async for us to wait for the answer. However, if you put it into a single command bundled with other tings (separated by `;`) it will run "sync" meaning the next one might start before the first one has completed.
